backend/backend.go

```diff
@@ -0,0 +1,12 @@
+package backend
+
+import "context"
+
+type Backend interface {
```

Comment:
Should we make these table methods? So the API will be `t.SetCursor(value string)`?

That would make it more restrictive, and avoid potential collisions. Also what's the reason we need a key? Can a single table have more than 1 key?

Reply:
Yeah, a single table can have more than 1 key. For example, if the table is multiplexed on account or project, we may need to pass in different IDs for each of those accounts/projects to resume from the last position. Generally I think only very simple cases will have a single key, most cases will have many.

Reply:
I like the table method idea! It might be hard to do in practice though. Right now resolvers don't have access to their own table, and even if they did, the table object wouldn't have a backend attached to it. I think we'll just have to accept that this will be the first pass, and we can iterate on the interface as we see how it's used in practice.

Reply:
>Yeah, a single table can have more than 1 key. For example, if the table is multiplexed on account or project, we may need to pass in different IDs for each of those accounts/projects to resume from the last position. Generally I think only very simple cases will have a single key, most cases will have many.

Multiplexed clients have an `ID()` method (I think tables can access it). Would that make a good key? I think that would make it easier to trace which cursor belongs to which client/table.

> I think we'll just have to accept that this will be the first pass

Good point. If we see the API is used in an inconsistent way we can change it (unless we want it that way :) )

Reply:
Yup, I haven't changed it to be a method on Table (which might be hard), but I have updated the Backend interface to be like this:

```
Set(ctx context.Context, table, clientID, value string) error
```

So basically we're relying on the client id to give uniqueness (which should be true if you use multiplexers correctly)